docs/extensibility/autocompletion.md

```diff
@@ -0,0 +1,85 @@
+Autocompletion
+==============
+
+Gutenberg provides a `blocks.Autocomplete.completers` filter for extending and overriding the list of autocompleters used by blocks.
+
+The `Autocomplete` component found in `@wordpress/blocks` applies this filter. The `@wordpress/components` package provides the foundational `Autocomplete` component that does not apply such a filter, but blocks should generally use the component provided by `@wordpress/blocks`.
+
+### Example
+
+Here is an example of using the `blocks.Autocomplete.completers` filter to add an acronym completer. You can find full documentation for the autocompleter interface with the `Autocomplete` component in the `@wordpress/components` package.
+
+{% codetabs %}
+{% ES5 %}
+```js
+// Our completer
+var acronymCompleter = {
+	name: 'acronyms',
```

Comment:
I've struggled with whether to make name singular or plural. It feels weird to call it "acronym" because it is not an acronym, it completes acronyms. I ended up using plural for the user and block completers and wanted to be consistent here. I'm fine changing it to singular everywhere though. Any thoughts?

Reply:
Let’s keep it consistent. No need to update  👍